File: NodeJS-Native/linearized-pdf-endpoint.js
/**
The /linearized-pdf endpoint can take a single PDF file or id as input.

This sample demonstrates linearizing a PDF file.
 */
// Importing the required modules
var https = require("follow-redirects").https;
var fs = require("fs");

// set the options for the HTTP request
var options = {
  method: "POST",
  hostname: "api.pdfrest.com",
  path: "/linearized-pdf",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Place your api key here
    "content-type":
      "multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW",
  },
  maxRedirects: 20,
};

// Sending the HTTPS POST request
var req = https.request(options, function (res) {
  var chunks = [];

  // Receiving data from the response
  res.on("data", function (chunk) {
    chunks.push(chunk);
  });

  // Handling the end of the response
  res.on("end", function (chunk) {
    var body = Buffer.concat(chunks);
    console.log(body.toString());
  });

  // Handling errors in the response
  res.on("error", function (error) {
    console.error(error);
  });
});

// set the data to be sent in the request body
var postData =
  '------WebKitFormBoundary7MA4YWxkTrZu0gW\r\nContent-Disposition: form-data; name="file"; filename="file"\r\nContent-Type: "{Insert_File_Content_Type}"\r\n\r\n' +
  fs.readFileSync("/path/to/file") +
  '\r\n------WebKitFormBoundary7MA4YWxkTrZu0gW\r\nContent-Disposition: form-data; name="output"\r\n\r\nlinearized_out\r\n------WebKitFormBoundary7MA4YWxkTrZu0gW--';

// write the data to the request body
req.write(postData);

// end the request
req.end();

// If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.js' sample.

File: Python/Single Calls/linearized-pdf-endpoint.py
from requests_toolbelt import MultipartEncoder
import requests
import json

linearized_pdf_endpoint_url = 'https://api.pdfrest.com/linearized-pdf'

# The /linearized-pdf endpoint can take a single PDF file or id as input.
# This sample demonstrates linearizing a PDF file.
mp_encoder_linearizedPdf = MultipartEncoder(
    fields={
        'file': ('toLinearized.pdf', open('../Sample_Input/toLinearize.pdf', 'rb'), 'application/pdf'),
        'output' : 'example_linearizedPdf_out',
    }
)

# Let's set the headers that the linearized-pdf endpoint expects.
# Since MultipartEncoder is used, the 'Content-Type' header gets set to 'multipart/form-data' via the content_type attribute below.
headers = {
    'Accept': 'application/json',
    'Content-Type': mp_encoder_linearizedPdf.content_type,
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' # place your api key here
}

print("Sending POST request to linearized-pdf endpoint...")
response = requests.post(linearized_pdf_endpoint_url, data=mp_encoder_linearizedPdf, headers=headers)

print("Response status code: " + str(response.status_code))

if response.ok:
    response_json = response.json()
    print(json.dumps(response_json, indent = 2))
else:
    print(response.text)

# If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.py' sample.
